control-plane/pkg/prober/async_prober.go

```diff
@@ -161,3 +165,28 @@ func (a *asyncProber) Probe(ctx context.Context, addressable Addressable, expect
 func (a *asyncProber) enqueueArg(_ string, arg interface{}) {
 	a.enqueue(arg.(types.NamespacedName))
 }
+
+// Resets the cache to a new empty cache with the given expiry time
+func (a *asyncProber) resetCache(ctx context.Context, expiry time.Duration) {
+	a.clientMu.Lock()
+	defer a.clientMu.Unlock()
+	a.cache = NewLocalExpiringCache(ctx, expiry)
+}
+
+func (a *asyncProber) RotateRootCaCerts(caCerts *string) error {
+	tlsClient := eventingtls.ClientConfig{CACerts: caCerts}
+	tlsClientConfig, err := eventingtls.GetTLSClientConfig(tlsClient)
+	if err != nil {
+		return err
+	}
+
+	a.clientMu.Lock()
+	defer a.clientMu.Unlock()
+	a.client = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsClientConfig,
+		},
+	}
```

Comment:
This code is very similar to the one in `New..TLS` above, can we refactor it to reduce duplication?